src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -436,22 +431,21 @@ private void handleException(Exception e) {
             long dataStartTime = request.getStart() - delayMillis;
             long dataEndTime = request.getEnd() - delayMillis;
 
-            Optional<Exception> previousException = stateManager.fetchExceptionAndClear(adID);
+            // HC logic starts here
+            if (anomalyDetector.isMultientityDetector()) {
+                Optional<Exception> previousException = stateManager.fetchExceptionAndClear(adID);
 
-            if (previousException.isPresent()) {
-                Exception exception = previousException.get();
-                LOG.error("Previous exception of {}: {}", adID, exception);
-                if (exception instanceof EndRunException) {
-                    EndRunException endRunException = (EndRunException) exception;
-                    if (endRunException.isEndNow()) {
-                        listener.onFailure(exception);
-                        return;
+                if (previousException.isPresent()) {
+                    Exception exception = previousException.get();
+                    LOG.error("Previous exception of {}: {}", adID, exception);
```

Comment:
This line will not print exception stack trace. Is this by design?

Reply:
Good catch.  Changed to 

LOG.error(new ParameterizedMessage("Previous exception of [{}]", adID), exception);